### PR TITLE
Categories Route

### LINKED ISSUE #85: Create categories route

### PR DESCRIPTION
**Pull request by Jordan141**
- [x] Add Categories.json
- [x] Add GET Categories route
- [x] Populate GET Categories route EJS with categories
- [x] Populate selectbox during article creation with categories and make it required
- [x] Extend Article Schema to accept categories
- [x] Verify desired category from client against server list of categories
- [x] Add list article by category route and appropriate EJS file

resolves #85 ,  #55, #49, #41

### CODE REVIEW

**Comment by Jordan141** [comment]
PR Done, image bug corrected in #116 is preventing appropriate testing, as such this PR is blocked from CR until said PR is merged.

Blocked by #116

File: app/models/article.js
Comment (michaelKurowski) [inline app/models/article.js:31]: I believe you can require JSON files in NodeJS. Like `require('./someJSON.json')`.

File: app/routes/articles.js
Comment (michaelKurowski) [inline app/routes/articles.js:15]: I believe you can require JSON files in NodeJS. Like `require('./someJSON.json')`.

File: app/routes/articles.js
Comment (michaelKurowski) [inline app/routes/articles.js:34]: Isn't body always present?

File: app/routes/index.js
Comment (michaelKurowski) [inline app/routes/index.js:32]: So we just let the response pending instead of redirecting to error page?

File: app/staticdata/categories.json
Comment (michaelKurowski) [inline app/staticdata/categories.json:1]: I think we should have an option to upload photos, because I wouldn't want to keep all these images in repo, but we still should keep them together with the app.

**Review by michaelKurowski — CHANGES_REQUESTED** [review]
What about choosing cats in edit article?

**Pull request merged**
